### PR TITLE
SubmitForm method

### PR DESCRIPTION
**Pull request by marklagendijk**
This pull requests implements the `response.submitForm` method, as discussed in #423. This pull request builds further on the work of #435.

The method takes the following parameters, all of which are optional:
- `formSelector` - the selector for finding the form in the response body. `form` by default.
- `submitSelector` - the selector for finding the submit button on the form. `[type="submit"]` by default.
- `fields` - key/value pairs for setting extra fields / overriding values for automatically set fields.
- `params` - custom request params which need to be applied to the request.

ToDo:
- [x] Write tests

### CODE REVIEW

**Comment by marklagendijk** [comment]
@liclac please review this when you have the time.

File: js/modules/k6/http/response.go [hunk 1]
```diff
@@ -141,3 +144,77 @@ func (res *HTTPResponse) Html(selector ...string) html.Selection {
...
 	return sel
 }
+
+func (res *HTTPResponse) SubmitForm(args ...goja.Value) (*HTTPResponse, error) {
+	rt := common.GetRuntime(res.ctx)
+
+	formSelector := "form"
+	submitSelector := "[type=\"submit\"]"
+	var fields map[string]goja.Value
+	var requestParams goja.Value
```
Comment (robingustafsson) [inline js/modules/k6/http/response.go]: If this line is changed to `requestParams := goja.Null()` we can remove the if/else at the end of the method when calling `Request()`.

File: js/modules/k6/http/response.go [hunk 2]
```diff
@@ -141,3 +144,77 @@ func (res *HTTPResponse) Html(selector ...string) html.Selection {
...
+	}
+
+	actionAttr := form.Attr("action")
+	var requestUrl goja.Value
+	if actionAttr == goja.Undefined() {
+		// Use the url of the response if no action is set
+		requestUrl = rt.ToValue(res.URL)
+	} else {
+		// Resolve the action url from the response url
+		responseUrl, _ := url.Parse(res.URL)
```
Comment (robingustafsson) [inline js/modules/k6/http/response.go]: Please add error checking to the two `url.Parse()` calls and call `common.Throw(...)` if there's an error.

File: js/modules/k6/http/response.go [hunk 3]
```diff
@@ -141,3 +144,77 @@ func (res *HTTPResponse) Html(selector ...string) html.Selection {
...
+
+	// Set the values supplied in the arguments, overriding automatically set values
+	for k, v := range fields {
+		body[k] = v
+	}
+
+	if requestParams == nil {
+		return New().Request(res.ctx, requestMethod, requestUrl, rt.ToValue(body))
+	} else {
+		return New().Request(res.ctx, requestMethod, requestUrl, rt.ToValue(body), requestParams)
```
Comment (robingustafsson) [inline js/modules/k6/http/response.go]: As mentioned in the other comment above this if/else can be reduced to just this line if `requestParams` initialized as a `goja.Null`.

**Pull request closed without merging**